src/components/userLists/index.jsx

```diff
@@ -1,49 +1,127 @@
 import "./style.css";
 import ProfileCircle from "../profileCircle/index.jsx";
 import Card from "../card/index.jsx";
-import EllipsisIcon from "../../assets/icons/ellipsisIcon.jsx";
+import { Navigate, NavLink, useNavigate } from "react-router-dom";
+import SimpleThreeDotsMenu from "../simpleThreeDotsMenu/index.jsx";
+import useUser from "../../hooks/useUser.jsx";
+import ThreeDotsMenu from "../threeDotsMenu/index.jsx";
+import { useEffect, useRef } from "react";
+import Button from "../button/index.jsx";
+import ListItem from "./listItem.jsx";
+
 const UserLists = ({ results, name }) => {
+  const { currentUser } = useUser();
+  const menuRef = useRef(null);
+  const navigate = useNavigate();
 
-    const getInitials = (firstName, lastName) => {
+  const getInitials = (firstName, lastName) => {
     const firstInitial = firstName ? firstName[0].toUpperCase() : "";
     const lastInitial = lastName ? lastName[0].toUpperCase() : "";
     return [firstInitial, lastInitial];
   };
 
-  const onClickStudent = (id) => {
-    setSelectedProfileId(id);
-    setIsStudentModalVisible(true);
+  const handleClickOutside = (event) => {
+    if (
+      menuRef.current &&
+      !menuRef.current.contains(event.target) &&
+      !event.target.closest(".link-to-profile")
+    ) {
+      setSelectedProfileId(null);
```

Comment:
Where is this setSelectedProfileId function defined? I can't see it in this file

Reply:
This function is used to remove the rendered profile popup after the three dot button is clicked, and the user clicks on another part of the screen. It's used in the onClickMenu to ensure when the profile is clicked it takes the user to that users profile page